### ISSUE TITLE
BasicShapeTemplateHarvester altering shape binding for View subdirectories

### ISSUE THREAD

**Issue by aodl** [OrchardCMS/OrchardCore#2640]
## Problem Overview

The Views folder can quickly become overloaded with templates, and it would be useful to be able to move these templates into appropriate subdirectories purely for organisational purposes. The default shape template harvester is already programmed to look in the following subdirectories; Items, Parts, Fields, Elements. However for the latter three, it alters the shape binding name by prefixing it with the subfolder name, meaning that the template will no longer bind to the same shape.

Currently, by default, the only subfolder that you can use without breaking the shape binding is `Views/Items` (as its a special case handled differently by the shape template harvester)

#### Current Implementation

`BasicShapeTemplateHarvester` searches in the following View sub-directories
- **Items** (templates in here do **not** have 'Items_' prefixed on the ShapeType string for shape/template binding)
- **Parts** (templates in here **do** have 'Parts_' prefixed on the ShapeType string for shape/template binding)
- **Fields** (templates in here **do** have 'Fields_' prefixed on the ShapeType string for shape/template binding)
- **Elements** (templates in here **do** have 'Elements_' prefixed on the ShapeType string for shape/template binding)

## Problem Statement

Given that the default template harvester is programmed to look in `Views/Parts` and `Views/Fields`, and also alter the shape binding name by prefixing `Parts_` or `Fields_` accordingly, why do these never get prefixed onto any shape names by the base part and field drivers?

### Solutions

It seems that `BasicShapeTemplateHarvester` should either be changed to treat the Views subdirectories in the same way that it currently treats `View/Items` (not prefixing the shape binding name), or the base part and field drivers should be altered to provide alternates that match the pattern applied by `BasicShapeTemplateHarvester`

**Comment by sebastienros** [comment]
> the same way that it currently treats View/Items

Right, the other behavior is some leakage from Orchard 1, where we named the parts differently. At the same time we can't take all folders as shapes, as it could lead to issues with other standard views. I think at some point we even looked into using a special folder, and @jtkech tried it, but it was too close to a release.

**Comment by jtkech** [comment]
> Right, the other behavior is some leakage from Orchard 1

Exactly, as i remember e.g `Parts.SomeShape.cshtml` is equivalent to `Parts/SomeShape.cshtml`.

Hmm, i think i tried to put a shape under `Items` and it was working but i don't remember if i had to add a prefix to find the shape.
